### PR TITLE
Add some safeguard to the test generation script.

### PR DESCRIPTION
**Pull request by fkling**
Just recording the parser output doesn't actually indicate whether the parser produces the correct output or not.
This adds at least a small safeguard by verifying that invalid tests throw an error and valid tests don't. Sections starting with "Invalid" are supposed to produce errors.

It actually found two problems:

> [XJS] Expected `<a .../*hai*/asdf/>` to result in an AST but got an error instead.
> [Invalid Type Annotations] Expected `var a: { [a: number]: string; [b: number]: string; };` to result in an error but got an AST instead.

cc @gabelevi @jeffmo

### CODE REVIEW

**Comment by gabelevi** [comment]
Yep, sounds like a good idea! Can you check why travis doesn't like it, though? I tried to re-submit the failing build to double check and it still failed.

**Comment by fkling** [comment]
Same issue as https://github.com/facebook/esprima/pull/86#issuecomment-71270117.

**Comment by jeffmo** [comment]
tldr of #86 is that Travis is having issues with node 0.11 -- but the tests are passing in other versions of node...so let's shipit!
